### PR TITLE
✨ Popup improvements

### PR DESCRIPTION
**Pull request by anuejn**
- ❌ close popups when they use focus
- 💅 animate popups

### CODE REVIEW

File: frontend/src/utils/use_blur.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useRef } from 'react';
+import { useEvent } from './use_event';
+
+export function useOnBlur(ref: HTMLElement | null, callback: () => void) {
+  // we need this terrible timeout hack because sometimes click events are fired, that are not on
+  // the target. these are immidiately followed by the actual event
```
Comment (pajowu) [inline frontend/src/utils/use_blur.ts:6]: How/when/why does this happen?

Comment (anuejn) [reply frontend/src/utils/use_blur.ts:6]: I am not really sure, why this happens, but it happens whenever I focus the slate editor and then try to operate a dropdown menu for speaker options

File: frontend/src/utils/use_state_delayed.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { useState } from 'react';
+
+export function useStateDelayed(
+  initial: boolean,
+  config = { late: 25, prolong: 200 },
+): [{ now: boolean; prolonged: boolean; late: boolean }, (next: boolean) => void] {
```
Comment (pajowu) [inline frontend/src/utils/use_state_delayed.ts:6]: Can i have a short comment what this does please? The variable names don't really help me understand it

File: frontend/src/utils/use_blur.ts (hunk 1 above)
Comment (pajowu) [reply frontend/src/utils/use_blur.ts:6]: Also this should probably be called `useDebouncedOnClick` or something, right?

Comment (anuejn) [reply frontend/src/utils/use_blur.ts:6]: Hm... not really, because it detects Clicks outside. maybe `useOnDebouncedOutsideClick`?

Comment (phlmn) [reply frontend/src/utils/use_blur.ts:6]: My dear fried, I hope your foot is well

https://github.com/transcribee/transcribee/blob/9704d73cea481f43ecca9cb65cb23ed3ae725deb/frontend/src/editor/transcription_editor.tsx#L122

Comment (phlmn) [reply frontend/src/utils/use_blur.ts]: > Also this should probably be called `useDebouncedOnClick` or something, right?

I vote for `useOnClickOutside`

**Pull request merged**